### DISCUSSION TITLE
Automatic static site generation with next-urql?

### DISCUSSION THREAD

**Discussion by tomduncalf** [urql-graphql/urql#2166] [Q&A] [answered]
Hey, thanks to the Urql team for a great library and I love how next-urql/withUrqlSsr easily enables SSR. 

I was wondering if it’s possible to make next-urql automatically output getStaticProps rather than getInitialProps, so the resulting build is fully static rather than server rendered?

The docs seems to imply you need to manually extract all your queries and write a getStaticProps for each page to do so. I naively tried swapping getInititalProps for getStaticProps in the withUrqlSsr code, but it blew up, lol. 

Thanks and happy holidays!
Tom

**Comment by JoviDeCroock** [comment, marked as answer]
GetStaticProps is a function you need to export so unless we write a compiler to do this it won't work 😅 this can't be achieved through runtime code

**Comment by tomduncalf** [comment]
Ahh I hadn’t made that distinction. Thanks for the explanation!
